lib/rack/timeout/legacy.rb

```diff
@@ -10,23 +10,24 @@ module Rack::Timeout::ClassLevelProperties
   module ClassMethods
     attr_accessor :service_timeout, :wait_timeout, :wait_overtime, :service_past_wait
     alias_method :timeout=, :service_timeout=
+  end
 
+  module CallWithDeprecation
     [ :service_timeout=,
       :timeout=,
       :wait_timeout=,
       :wait_overtime=,
       :service_past_wait=,
     ].each do |isetter|
-      setter = instance_method(isetter)
-      define_method(isetter) do |x|
+      define_method(isetter) do |*|
         defined?(Rails) or warn "`Rack::Timeout.#{isetter}`: class-level settings are deprecated. See README for examples on using the middleware initializer instead."
-        setter.bind(self).call(x)
+        super
```

Comment:
Since I'm not 100% sure about the compatibility, I wrote a test code and it seems the behavior is same:

```ruby
module M1
  def f
    p 1
  end
end

module N1
  def f
    p 2
    super
  end
end

M1.prepend N1

class A
  extend M1
end

A.f
# 2
# 1

module M2
  def f
    p 1
  end

  setter = instance_method(:f)
  define_method(:f) do
    p 2
    setter.bind(self).call
  end
end

class A
  extend M2
end

A.f
# 2
# 1
```